test/fuzz/parser_fuzzers.go

```diff
@@ -0,0 +1,337 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fuzz
+
+import (
+	"io"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/util/keyutil"
+	"k8s.io/code-generator/cmd/client-gen/generators/util"
+	envutil "k8s.io/kubectl/pkg/cmd/set/env"
+	"k8s.io/kubectl/pkg/util/certificate"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"k8s.io/kubernetes/pkg/kubelet/eviction"
+	"k8s.io/kubernetes/pkg/registry/core/service/portallocator"
+	"k8s.io/kubernetes/pkg/util/parsers"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+)
+
+// FuzzParseSelector implements a fuzzer
+// that targets fields.ParseSelector
+func FuzzParseSelector(data []byte) int {
+	_, _ = fields.ParseSelector(string(data))
+	return 1
+}
+
+// FuzzParseQuantity implements a fuzzer
+// that targets resource.ParseQuantity
+func FuzzParseQuantity(data []byte) int {
+	_, _ = resource.ParseQuantity(string(data))
+	return 1
```

Comment:
Only return 1 if the data parsed successfully. If the function returns an error, you should return 0.

Reply:
The reason I do this is from a suggestion by one of the go-fuzz contributors. See [this comment](https://github.com/tailscale/tailscale/pull/1115#discussion_r556171316):
"*don't worry about returning 0 vs 1. The signal from those is weak, and go-fuzz's support for running multiple fuzz functions is poor enough that it's nicer to just have a single mixed fuzz function*"

Am happy to do either or, so let me know which you prefer.